### ISSUE TITLE
Unlit furnace: this block is no longer here

### ISSUE THREAD

**Issue by SquidDev** [SquidDev-CC/ccjam-2016#34]
> Drop wood into an unlit furnace(that has coal in it) with this script:
```lua
function printname()
  for k, v in pairs(furnace.getMetadata()) do
    if k == "name" then
    print(v)
    end
  end
end
furnace = peripheral.wrap("bottom")
printname()
turtle.dropDown(1)
furnace = peripheral.wrap("bottom")
printname()
```
> This works the first time but when trying to repeat it just returns "The block is no longer there"
I circumvented it by having the turtle go up and down before the second wrap but thought you'd might want to look into it.

**Comment by SquidDev** [comment]
This is probably caused by the [block state changing](https://github.com/SquidDev-CC/plethora/blob/minecraft-1.8.9/src/main/java/org/squiddev/plethora/api/reference/BlockReference.java#L39) (furnace lit vs unlit). We should at least compare the block instead of the state, and perhaps ignore the entire state if we have a tile entity.

**Comment by SquidDev** [comment]
~~It appears this is actually because the tile itself changes. I'm not sure of an elegant way to solve this, maybe custom equality handler for blocks and tiles.~~

~~**Edit:** That wouldn't work as the capabilities would still be pointing to the old tile.~~

Even worse: the block change is triggered with a new tile, then then the tile is reset to the original version. As the block isn't changed, the peripheral isn't updated. It might be worth seeing what OpenPeripheral does in this instance.

**Comment by SquidDev** [comment]
This is also a bug in OpenPeripheral (OpenMods/OpenPeripheral#198) if it makes me feel less bad. Not sure what an ideal solution is 😦.

**Comment by SquidDev** [comment]
So a temporary solution is just to use modems as they listen to a block being removed and replaced rather than changed. The latest commit ensures modems work correctly which is nice.

**Comment by SquidDev** [comment]
This is sort of fixed by dan200/ComputerCraft#180. It isn't perfect as the peripheral will be reset each time, but there isn't much more we can do.
